frontend/app/src/components/history/transactions/TransactionEvents.vue

```diff
@@ -0,0 +1,175 @@
+                  :text="note.word"
+                  :tx="note.isTransaction"
+                />
+              </span>
+              <span v-else :key="index">
+                {{ note.word }}
+              </span>
+            </template>
+          </template>
+        </data-table>
+      </div>
+    </template>
+  </table-expand-container>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  PropType,
+  toRefs,
+  unref
+} from '@vue/composition-api';
+import { DataTableHeader } from 'vuetify';
+import TableExpandContainer from '@/components/helper/table/TableExpandContainer.vue';
+import TransactionEventAsset from '@/components/history/transactions/TransactionEventAsset.vue';
+import TransactionEventTypeWrapper from '@/components/history/transactions/TransactionEventTypeWrapper.vue';
+import i18n from '@/i18n';
+import { EthTransactionEventWithMeta } from '@/services/history/types';
+import {
+  EthTransactionEntry,
+  EthTransactionEventEntry
+} from '@/store/history/types';
+import { transformEntryWithMeta } from '@/store/history/utils';
+
+const headers: DataTableHeader[] = [
+  {
+    text: i18n.t('transactions.event_headers.type').toString(),
+    value: 'type',
+    sortable: false,
+    cellClass: 'type pl-0'
+  },
+  {
+    text: i18n.t('transactions.event_headers.asset').toString(),
+    value: 'asset',
+    sortable: false
+  },
+  {
+    text: i18n.t('transactions.event_headers.description').toString(),
+    value: 'description',
+    sortable: false,
+    width: '50%',
+    cellClass: 'description pr-0'
+  }
+];
+
+export default defineComponent({
+  name: 'TransactionEvents',
+  components: {
+    TransactionEventTypeWrapper,
+    TransactionEventAsset,
+    TableExpandContainer
+  },
+  props: {
+    transaction: {
+      required: true,
+      type: Object as PropType<EthTransactionEntry>
+    },
+    colspan: { required: true, type: Number }
+  },
+  setup(props) {
+    const { transaction } = toRefs(props);
+
+    const events = computed<EthTransactionEventEntry[]>(() => {
+      return unref(transaction).decodedEvents!.map(
+        (event: EthTransactionEventWithMeta) => {
+          return transformEntryWithMeta(event);
+        }
+      );
+    });
+
+    const formatNotes = (
+      notes: string
+    ): { word: string; isAddress: boolean; isTransaction: boolean }[] => {
+      // label each word from notes whether it is an address or not
+      const words = notes.split(' ');
+
+      return words.map((word, index) => {
+        const isAddress = word.startsWith('0x') && word.length >= 42;
+        const isTransaction =
+          isAddress && index !== 0 && words[index - 1] === 'transaction';
+        return {
+          word,
+          isAddress,
+          isTransaction
+        };
+      });
+    };
+
+    return {
+      formatNotes,
+      events,
+      headers
+    };
+  }
+});
+</script>
+
+<style scoped lang="scss">
+::v-deep {
```

Comment:
Which component is getting overwritten here?

Reply:
If it is our own component then maybe it makes to add this in the component itself